tests/multiple_validator_set_updates_test.rs

```diff
@@ -0,0 +1,121 @@
+use std::{thread, time::Duration};
+
+use log::LevelFilter;
+use rand_core::OsRng;
+
+use hotstuff_rs::types::{
+    basic::{AppStateUpdates, Power},
+    collectors::SigningKey,
+    validators::ValidatorSetUpdates,
+};
+
+mod common;
+
+use common::{
+    logging::setup_logger,
+    network::mock_network,
+    node::Node,
+    number_app::{NumberAppTransaction, NUMBER_KEY},
+};
+
+#[test]
+fn multiple_validator_set_updates_test() {
```

Comment:
This test _sometimes_ (~20% of runs) does not complete when run on my mac. In these non-terminating scenarios, the replicas continue to work and print out log messages, but execution never breaks out of the while loop in step 4.2.

Here are the logs generated during one of these non-terminating runs: [multiple_validator_set_updates_test.log](https://github.com/user-attachments/files/15905797/multiple_validator_set_updates_test.log)

Reply:
Turns out the 20% figure is a big overestimate. The logical conditions that the while loops in this test check were inverted. They were equivalent to "while all replicas' CVSs have N validators", whereas they should have been "while it is not the case that all replicas' CVSs have N validators". This caused the test code to proceed all the way to the last while loop almost immediately, as reflected in the logs I posted. Once at this final while loop, which continues "while all replicas' CVSs have 2 validators", the execution would often get stuck, because the final CVS in the test case has exactly 2 validators.

This problem is fixed in [d3f93d5](https://github.com/parallelchain-io/hotstuff_rs/pull/36/commits/d3f93d53d66bb4e78828f30c1c917dd72fb78046).